src/libsyntax/codemap.rs

```diff
@@ -578,6 +583,17 @@ impl CodeMap {
         }
     }
 
+    /// Return the source snippet as `String` corresponding to the given `Span`
+    pub fn span_to_snippet(&self, sp: Span) -> Result<String, SpanSnippetError> {
+        self.span_to_source(sp, |src, start_index, end_index| src[start_index..end_index]
+                                                                .to_string())
+    }
+
+    /// Return the source snippet as `String` before the given `Span`
```

Comment:
Can you expand this comment a bit -- where does the resulting string start? From the beginning of the containing file?

Reply:
I'm not actually sure. It comes from either the `src` or the `external_src` field of the `FileMap` associated with the span. I think that this is the containing file, but comments like "[...] source that doesn't originate from files [...]" in the `FileMap` struct make me doubt myself.

Should I stay vague and put something like: "The snippet will begin at the start of the codemap's source associated with the span.", or should I specifically say "The snippet will begin at the start of the file associated with the span."?

Reply:
maybe just leave a comment saying you are not sure :)